pyam/core.py

```diff
@@ -113,7 +113,7 @@ def __getattr__(self, attr):
 
         if attr in self.dimensions:
             ret = self._iamcache[attr] = self.index[self].unique(level=attr).tolist()
-            return ret
+            return pd.Index(ret) if attr == "time" else ret
 
```

Comment:
Ah i was not aware that the time representation was returning proper index types instead of lists. Let's save the indices then, instead of round-robin converting:
```python
    def __getattr__(self, attr):
        ret = object.__getattribute__(self, "_iamcache").get(attr)
        if ret is not None:
            return ret.tolist() if attr != "time" else ret

         if attr in self.dimensions:
            ret = self._iamcache[attr] = self.index[self].unique(level=attr)
            return ret.tolist() if attr != "time" else ret
```